src/Mvc/Mvc.Core/src/Authorization/AuthorizeFilter.cs

```diff
@@ -169,20 +160,9 @@ internal async Task<AuthorizationPolicy> GetEffectivePolicyAsync(AuthorizationFi
                 {
                     builder.Combine(endpointPolicy);
                 }
-
-                // We cannot cache the policy since it varies by endpoint metadata.
-                canCache = false;
-            }
-
-            effectivePolicy = builder?.Build() ?? effectivePolicy;
-
-            // We can cache the effective policy when there is no custom policy provider
-            if (canCache)
-            {
-                _effectivePolicy = effectivePolicy;
             }
 
-            return effectivePolicy;
+            return builder?.Build() ?? effectivePolicy;
```

Comment:
Am I crazy? It doesn't look like it's actually possible for `builder` to be null here.

Reply:
Yeah looks like I can simplify this method a bit, I'll clean it up